### PR TITLE
Let Attention layer accept a layer mask of list of length 3, namely [query_mask, value_mask, key_mask] #41294

### PR DESCRIPTION
**Pull request by bubblebooy**
#41294 

First pull request feel free to let me know if I am doing anything wrong. Thanks

for tf.keras.layers.Attention currently the layer mask must be a list of length 2, namely [query_mask, value_mask]
change it to also accept a layer mask of list of length 3, namely [query_mask, value_mask, key_mask]

currently def compute_mask() returns the query_mask and ignores the value_mask. If passed a key_mask it could also just be ignored

### CODE REVIEW

**Comment by gbaned** [comment]
@bubblebooy Can you please sign CLA and resolve conflicts? 
Can you also use a relevant title for the PR and include a short description in the first message field?  Thank you!

**Comment by bubblebooy** [comment]
@googlebot I consent.

**Comment by joker-eph** [comment]
Seems like this branch includes many TensorFlow commits rebased weirdly on master, I suspect it needs some cleanup.
 Looking at the "commits" tab you need to make sure it only have your own commits.

File: tensorflow/python/keras/layers/dense_attention.py [hunk 1]
```diff
@@ -192,10 +192,10 @@ def _validate_call_args(self, inputs, mask):
         raise ValueError(
             '{} layer mask must be a list, '
             'namely [query_mask, value_mask].'.format(class_name))
-      if len(mask) != 2:
+      if len(mask) < 2 or len(mask) > 3:
         raise ValueError(
-            '{} layer mask must be a list of length 2, namely [query_mask, '
-            'value_mask]. Given length: {}'.format(class_name, len(mask)))
+            '{} layer mask must be a list of length 2 or 3, namely [query_mask, '
+            'value_mask] or [query_mask, value_mask, key_mask]. Given length: {}'.format(class_name, len(mask)))
```
Comment (qlzh727) [inline tensorflow/python/keras/layers/dense_attention.py:198]: Please don't change it to include key_mask, since we don't expect user to provide it, and will not be used even provided.

Comment (bubblebooy) [reply tensorflow/python/keras/layers/dense_attention.py:198]: My intent was for it not to error when 3 masks are passed via implicit propagation as in the example below. Should I remove the changes to the ValueError() while still letting it accept 3 arguments or just close this pull request?

```
input = tf.keras.layers.Input(shape=(None,))
x = tf.keras.layers.Embedding(input_vocab_size, d_model, mask_zero=True)(input)
query = tf.keras.layers.Dense(d_model)(x)
value =  tf.keras.layers.Dense(d_model)(x)
key    =  tf.keras.layers.Dense(d_model)(x)
attention = tf.keras.layers.Attention()([query, value, key])
...
```

Comment (qlzh727) [reply tensorflow/python/keras/layers/dense_attention.py:198]: I think you can keep the check for len(mask) in (2, 3), but change the error message back to only list the query_mask and value_mask. We don't want to encourage user to explicitly pass in the key_mask, which they can do like:

```
attention = tf.keras.layers.Attention()([query, value, key], 
    masks=[query._keras_mask, value._keras_mask, key._keras_mask])
```

**Review by qlzh727 — DISMISSED** [review]

**Review by qlzh727 — APPROVED** [review]
